lib/kill.js

```diff
@@ -65,6 +81,9 @@ export const validateTimeout = ({timeout}) => {
 	}
 };
 
+// Indicates that the error is used only to interrupt control flow, but not in the return value
+const setDiscardedError = error => Object.assign(error, {discarded: true});
```

Comment:
For timeouts, non-0 exit codes and signal terminations, we discard the original error message and stack trace. Instead, we let the main makeError() logic set those later on.